README.md

```diff
@@ -1,22 +1,336 @@
+$(function () {
+  $(".js-more").on('click', function() {
+    var id = $(this).data("id");
+    // change the URL to the new route
+    $.get("/posts/" + id + "/post_data", function(data) {
+      // Replace text of body-id div
+      $("#body-" + id).text(data);
+    });
+  });
+});
+</script>
+```
+
+Okay, now if we reload and try again, we can see that our button now
+replaces the truncated body with the JSON. This makes sense. Before we
+were just returning a single value, but now we have a whole object. So
+we'll need to alter the code to just pull the `description` from the
+response JSON.
+
+```erb
+# posts/index.html.erb
+# ...
+<script type="text/javascript" charset="utf-8">
+$(function () {
+  $(".js-more").on('click', function() {
+    var id = $(this).data("id");
+    $.get("/posts/" + id + "/post_data", function(data) {
+      // Replace text of body-id div
+      $("#body-" + id).html(data["description"]);
+    });
+  });
+});
+</script>
+```
+
+As you can see, we can access the JSON just like any other dictionary,
+and get the value for the `"description"` key.
+
+### Using JSON To Build The Show Page
+
+Now we want to set up our blog so that we can click a link to see the
+next entry when we're doing reading the current one, but we don't want
+to have to refresh the page.
+
+We can reuse our `post_data` API, since that's exactly what a `Post`
+show page should have on it!
+
+We need to update the HTML part of our template with some metadata we
+can use to identify the different elements, and add our `Next` link.
+
+```erb
+# posts/show.html.erb
+<div class="authorName"><%= @post.author.name %></div>
+<a href="#" class="js-next" data-id="<%=@post.id%>">Next...</a>
+<h1 class="postTitle"><%= @post.title %></h1>
+<p class="postBody"><%= @post.description %></p>
+```
+
+Note that we are using `data-id` on the `<a>`. We'll need to keep that
+updated so that we know what post to request next.
+
+Okay, now let's add the JavaScript to handle this.
+
+```erb
+# posts/show.html.erb
+# ...
+<script type="text/javascript" charset="utf-8">
+$(function () {
+  $(".js-next").on("click", function() {
+    var nextId = parseInt($(".js-next").attr("data-id")) + 1;
+    $.get("/posts/" + nextId + "/post_data", function(data) {
+      $(".authorName").text(data["author"]["name"]);
+      $(".postTitle").text(data["title"]);
+      $(".postBody").text(data["description"]);
+      // re-set the id to current on the link
+      $(".js-next").attr("data-id", data["id"]);
+    });
+  });
+});
+</script>
+```
+
+Here we grab the `data-id` attribute value, use `parseInt()` so we can
+cast it and add one to it so we can request the *next* post.
+
+After that, it's just a matter of using `$.get()` to hit our `post_data`
+endpoint, and replace the `text` of each par of our show page.
+
+If we reload and go to our first post, then click `Next`, we get that
+smooth AJAX data refresh we've been looking for.
+
+At least until we get to the last post.
```

Comment:
I think this lesson is great overall. The challenge, as you pointed out, will be in balancing helping the student out/setting them up to understand more advanced concepts versus holding their hand and making the code-along too easy.